From 4450530803e7ba1dea0153ef81ad2198f80c991f Mon Sep 17 00:00:00 2001
From: Joseph Groenenboom <jgroenen@redhat.com>
Date: Tue, 15 Oct 2024 16:26:46 -0400
Subject: [PATCH 4/4] Remove download from build

Allow the build to skip scm/git checkouts and build as it did in v0.6.1.

Specify target version name so we can correctly build the version of vLLM we need.

Below we set __version__ as part of the build workaround. This may be changed in the 
future but for now it allows us to get around a check that used to be used to stage a
checkout but we still want to use a tarball.
---
 setup.py        | 15 ++++++++++++---
 vllm/version.py |  2 +-
 2 files changed, 13 insertions(+), 4 deletions(-)

diff --git a/setup.py b/setup.py
index 8ef759f5..f6f1830b 100644
--- a/setup.py
+++ b/setup.py
@@ -352,11 +352,20 @@ def get_nvcc_cuda_version() -> Version:
 def get_path(*filepath) -> str:
     return os.path.join(ROOT_DIR, *filepath)
 
+def find_version(filepath: str) -> str:
+    """Extract version information from the given filepath.
+
+    Adapted from https://github.com/ray-project/ray/blob/0b190ee1160eeca9796bc091e07eaebf4c85b511/python/setup.py
+    """
+    with open(filepath) as fp:
+        version_match = re.search(r"__version__ = ['\"]([^'\"]*)['\"]",
+                                  fp.read(), re.M)
+        if version_match:
+            return version_match.group(1)
+        raise RuntimeError("Unable to find version string.")
 
 def get_vllm_version() -> str:
-    version = get_version(
-        write_to="vllm/_version.py",  # TODO: move this to pyproject.toml
-    )
+    version = find_version(get_path("vllm", "version.py"))
 
     sep = "+" if "+" not in version else "."  # dev versions might contain +
 
diff --git a/vllm/version.py b/vllm/version.py
index 66e189dc..b8ee1735 100644
--- a/vllm/version.py
+++ b/vllm/version.py
@@ -7,5 +7,5 @@ except Exception as e:
                   RuntimeWarning,
                   stacklevel=2)
 
-    __version__ = "dev"
+    __version__ = "0.6.2"
     __version_tuple__ = (0, 0, __version__)
-- 
2.43.5

